clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -115,18 +114,44 @@ public boolean getOnlySpinnakerManaged() {
 
   private final Path serviceAccountNamespacePath = Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/namespace");
 
+  public enum InvalidKindReason {
+    KIND_NONE("Kind [%s] is invalid"),
+    EXPLICITLY_OMITTED_BY_CONFIGURATION("Kind [%s] included in 'omitKinds' of kubernetes account configuration"),
+    MISSING_FROM_ALLOWED_KINDS("Kind [%s] missing in 'kinds' of kubernetes account configuration"),
+    READ_ERROR("Error reading kind [%s]. Please check connectivity and access permissions to the cluster");
+
+    private String errorMessage;
+    InvalidKindReason(String errorMessage) {
+      this.errorMessage = errorMessage;
+    }
+
+    public String getErrorMessage(KubernetesKind kind) {
+      return String.format(this.errorMessage, kind);
+    }
+  }
+
   public boolean isValidKind(KubernetesKind kind) {
```

Comment:
Could this be refactored now to just `return getInvalidKindReason(kind) != null`?  Just to make sure the two functions don't diverge?

Reply:
That sounds good!